ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebView.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ * <p/>
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.views.webview;
+
+import android.annotation.SuppressLint;
+import android.text.TextUtils;
+import android.webkit.WebView;
+
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.uimanager.ThemedReactContext;
+
+import javax.annotation.Nullable;
+
+/**
+ * Subclass of {@link WebView} that implements {@link LifecycleEventListener} interface in order
+ * to call {@link WebView#destroy} on activity destroy event and also to clear the client
+ */
+@SuppressLint("ViewConstructor")
+public class ReactWebView extends WebView implements LifecycleEventListener {
```

Comment:
@ide I pulled this class out of `ReactWebViewManager` and made it public so that I could spy on it using Mockito and test the functionality of the ReactWebViewManager. 

I figured other simliar classes like this are already public (eg. https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java et al.) so there is no harm